Sources/PaymentsCore/AnalyticsEventRequest.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+
+struct AnalyticsEventRequest: APIRequest {
+    
+    init(payload: AnalyticsPayload) throws {
+        let encoder = JSONEncoder()
+        encoder.keyEncodingStrategy = .convertToSnakeCase
+        body = try encoder.encode(payload)
+    }
+    
+    // MARK: - APIRequest
+    
+    typealias ResponseType = EmptyResponse
```

Comment:
Just a general question, is this typealias being uses or is it for future implementation

Reply:
Due to how our [`APIRequest` protocol ](https://github.com/paypal/iOS-SDK/blob/main/Sources/PaymentsCore/Networking/Protocols/APIRequest.swift)was designed, it is required to implement an associated, Decodable, response type. So even though this API call has no return details

Doing this PR pointed out a handful of limitations of our current networking layer. Matt and I's plan was to make this initial PR as-is, without changing any of our fundamental networking logic, and just fitting the mold of what currently exists. And then in a follow-up PR, when time permits, make broader changes to the `APIRequest` protocol (or removing it entirely).

For example another weird thing we had to do is on lines 21-23 of this file, to override the environment URL.

Reply:
Gotcha - I think a follow up makes a lot of sense for this! Good for us to fins these limitations now so we can improve the networking layer. 🚀